codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamFrameToHttpObjectCodec.java

```diff
@@ -268,4 +268,20 @@ private static boolean isInformationalResponseHeaderFrame(CharSequence status) {
         }
         return false;
     }
+
+    /*
+     * https://datatracker.ietf.org/doc/html/rfc9113#section-8.1.1
+     * '204' or '304' responses contain no content
+     */
+    private static boolean isContentAlwaysEmpty(CharSequence status) {
+        if (status.length() == 3) {
+            char char0 = status.charAt(0);
+            char char1 = status.charAt(1);
+            char char2 = status.charAt(2);
+            return char0 >= '2' && char0 <= '3'
```

Comment:
```suggestion
            return (char0 == '2' || char0 == '3')
```

Reply:
You can check what I have done for Netty 4.1 in a recent pr and use a static final lookup table.

Reply:
fixed